app/PermissionsContext.js

```diff
@@ -0,0 +1,141 @@
+import React, { createContext, useEffect, useState } from 'react';
```

Comment:
An alternative to the context could have been some custom hooks. I think the idea of a context for this makes sense though since it seems like there are a lot of desperate screens/components that want to be interacting with react-native-permissions and a context is a good way to keep a single source of truth for this situation.

That said, I'm not sure what the plan is for permission management is in general, so please let me know if this is butting up against other changes in the works and i can update it follow the preferred pattern.

Finally, while i think its best practice to not have all this conflated with the view logic, as a worst case scenario if this conflicts with other work to a large degree, i can inline this logic back into the screen and we can find a way to pull it out more holistically in a latter commit.

Reply:
Agreed on this not being in view logic. A context seems fine for now. 


One thing to note, we are already checking location permissions in the `LocationServices.checkStatus()`. It uses a slightly different method than you have here 

(Hopefully they yield the same results.. but they're technically 2 different modules).

Probably no action here, but wanted to make you aware.

Reply:
got it, havent had a chance to dig into the location service, so dont have any strong opinions, but it might makes sense to consolidate and have the location service defer to the permissions context for permission stuff.

Reply:
Thanks for the write up!

> To summarize - I think both this file and LocationService are providing two slightly different sets of permission info. This ContextProvider is purely app-specific device permissions, whereas LocationService also takes into account device and user-level statuses.

This makes a lot of sense to me as a distinction.